### PR TITLE
MULE-8817: Upgrade Spring to 4.3.16

### PR DESCRIPTION
**Pull request by aiannucci**

### CODE REVIEW

File: modules/cxf/src/test/java/org/mule/module/cxf/CxfInboundMessageProcessorTestCase.java [hunk 1]
```diff
@@ -77,11 +77,9 @@ public MuleEvent process(MuleEvent event) throws MuleException
         MuleEvent event = getTestEvent(msg, getTestInboundEndpoint(REQUEST_RESPONSE));
         
         MuleEvent response = processor.process(event);
-        
         Object payload = response.getMessage().getPayload();
-        assertTrue(payload instanceof OutputHandler);
         
-        ((OutputHandler) payload).write(response, System.out);
```
Comment (elrodro83) [inline modules/cxf/src/test/java/org/mule/module/cxf/CxfInboundMessageProcessorTestCase.java]: why was this removed?

File: modules/spring-config/pom.xml [hunk 2]
```diff
@@ -69,12 +27,6 @@
             <artifactId>spring-context</artifactId>
         </dependency>
 
-        <dependency>
-            <groupId>cglib</groupId>
-            <artifactId>cglib</artifactId>
-            <version>${springCglibVersion}</version>
```
Comment (elrodro83) [inline modules/spring-config/pom.xml]: remove this property from the root pom if it is not used elsewhere

File: modules/spring-extras/src/main/java/org/mule/module/spring/events/MuleEventMulticaster.java [hunk 3]
```diff
@@ -412,6 +413,17 @@ else if (!(listener instanceof MuleEventListener))
...
     }
 
+    /***
+     * Implemented only for compliance with Spring 4.2+
+     * Please use the {@link #multicastEvent(ApplicationEvent) multicastEvent} method.
+     * @param event
+     * @param eventType
+     */
+    @Override
+    public void multicastEvent(ApplicationEvent event, ResolvableType eventType) {
```
Comment (elrodro83) [inline modules/spring-extras/src/main/java/org/mule/module/spring/events/MuleEventMulticaster.java]: is this related to the spring upgrade?

File: modules/spring-config/pom.xml (hunk 2 above)
Comment (aiannucci) [reply modules/spring-config/pom.xml]: It was declared in this POM on line 17, removed as part of this commit.

File: modules/spring-extras/src/main/java/org/mule/module/spring/events/MuleEventMulticaster.java (hunk 3 above)
Comment (aiannucci) [reply modules/spring-extras/src/main/java/org/mule/module/spring/events/MuleEventMulticaster.java]: Yes, that's why I mention in the Javadoc that we must implement this for compliance with the Spring 4.2+ ApplicationEventMulticaster interface that this class implements.

File: modules/cxf/src/test/java/org/mule/module/cxf/CxfInboundMessageProcessorTestCase.java (hunk 1 above)
Comment (aiannucci) [reply modules/cxf/src/test/java/org/mule/module/cxf/CxfInboundMessageProcessorTestCase.java]: We shouldn't write to sysout directly and this had no purpose. I added an assertion on the response message instead.
